esmvaltool/diag_scripts/mlr/__init__.py

```diff
@@ -0,0 +1,837 @@
+    output = True
+    accepted_modes = ('full', 'only_missing', 'only_var_type')
+    if mode not in accepted_modes:
+        raise ValueError(
+            f"'mode' must be one of {accepted_modes}, got '{mode}'")
+    for dataset in datasets:
+        if mode != 'only_var_type':
+            for key in NECESSARY_KEYS:
+                if key not in dataset:
+                    getattr(logger, log_level)(
+                        "Dataset '%s' does not have necessary (MLR) attribute "
+                        "'%s'", dataset, key)
+                    output = False
+        if mode != 'only_missing' and dataset.get('var_type') not in VAR_TYPES:
+            getattr(logger, log_level)(
+                "Dataset '%s' has invalid var_type '%s', must be one of %s",
+                dataset, dataset.get('var_type'), VAR_TYPES)
+            output = False
+    return output
+
+
+def get_1d_cube(x_data, y_data, x_kwargs=None, y_kwargs=None):
+    """Convert 2 arrays to :class:`iris.cube.Cube` (with single coordinate).
+
+    Parameters
+    ----------
+    x_data : numpy.ndarray
+        Data for coordinate.
+    y_data : numpy.ndarray
+        Data for cube.
+    x_kwargs : dict
+        Keyword arguments passed to :class:`iris.coords.AuxCoord`.
+    y_kwargs : dict
+        Keyword arguments passed to :class:`iris.cube.Cube`.
+
+    Returns
+    -------
+    iris.cube.Cube
+        1D cube with single auxiliary coordinate.
+
+    Raises
+    ------
+    ValueError
+        Arrays are not 1D and do not have matching shapes.
+
+    """
+    if x_kwargs is None:
+        x_kwargs = {}
+    if y_kwargs is None:
+        y_kwargs = {}
+    x_data = np.ma.array(x_data)
+    y_data = np.ma.array(y_data)
+    if x_data.ndim != 1:
+        raise ValueError(
+            f"Expected 1D array for 'x_data', got {x_data.ndim:d}D array")
+    if y_data.ndim != 1:
+        raise ValueError(
+            f"Expected 1D array for 'y_data', got {y_data.ndim:d}D array")
+    if x_data.shape != y_data.shape:
+        raise ValueError(
+            f"Expected identical shapes for 'x_data' and 'y_data', got "
+            f"{x_data.shape} and {y_data.shape}, respectively")
+    aux_coord = iris.coords.AuxCoord(x_data, **x_kwargs)
+    cube = iris.cube.Cube(y_data, aux_coords_and_dims=[(aux_coord, 0)],
+                          **y_kwargs)
+    return cube
+
+
+def get_absolute_time_units(units):
+    """Convert time reference units to absolute ones.
+
+    This function converts reference time units (like ``'days since YYYY'``) to
+    absolute ones (like ``'days'``).
+
+    Parameters
+    ----------
+    units : cf_units.Unit
+        Time units to convert.
+
+    Returns
+    -------
+    cf_units.Unit
+        Absolute time units.
+
+    Raises
+    ------
+    ValueError
+        If conversion failed (e.g. input units are not time units).
+
+    """
+    if units.is_time_reference():
+        units = Unit(units.symbol.split()[0])
+    if not units.is_time():
```

Comment:
This is a good question, I honestly do not know, but I haven't seen other cases.

Reply:
aha so it's a 'if it ain't broken don't fix it' thing...ok, if it works here, let it be then

Reply:
note that eg `cf_units.Unit("kg s-1").symbol.split()[0]` will return `kg.s-1` so am not sure why you doing the splitting and indexing though...meh, let's not `split` hairs, if this works for you OK

Reply:
It's only meant for time units (if used with other units a `ValueError` is raised). The use of this function is e.g. to convert from `days since 1850-01-01` to `days`.